pkg/compiler/scope.go

```diff
@@ -53,7 +62,7 @@ func (s *scope) SetVariable(name string) error {
 	}
 
 	// TODO: add type detection
```

Comment:
There is no types now

Reply:
Todo for the future. Until we implement it, no reason to use type as a map value.